### PR TITLE
Replace token flags with = to prevent bad parsing of leading dash in token

### PR DESCRIPTION
**Pull request by candrewlee14**
This is a pretty awkward problem, and it stems from the fact that those wrapping quotes are parsed away when Go gets the program's arguments for `os.Args`.

I ran `dlv debug ./cmd/influx -- config create -n targetflux -u "http://localhost:8087" -o "organization" -t "-therestofmytoken"`
And this is the `os.Args` value which `urfave/cli` uses directly.
```
(dlv) p os.Args
[]string len: 11, cap: 11, [
        "...+2 more",
        "config",
        "create",
        "-n",
        "targetflux",
        "-u",
        "http://localhost:8087",
        "-o",
        "organization",
        "-t",
        "-therestofmytoken",      <--- no wrapping quotes
]
```
It seems like a bespoke method specifically for replacing `-t "-FOO-TOKEN"` with `-t=-FOO-TOKEN` in `os.Args` before passing into `urfave/cli` makes the most sense here.

Closing #376 and #378

### CODE REVIEW

File: cmd/influx/main.go [hunk 1]
```diff
@@ -91,8 +92,31 @@ func newApp() cli.App {
...
+			} else {
+				os.Args = os.Args[:i+1]
+			}
+		}
+	}
+}
+
 func main() {
 	app := newApp()
+	replaceTokenArg()
```
Comment (lesam) [inline cmd/influx/main.go:119]: This is clever! Can we create a new slice instead of messing with the os.Args one though (and pass that to `app.Run`?)

**Comment by samhld** [comment]
Will both options still be available from the user perspective?

**Comment by candrewlee14** [comment]
> Will both options still be available from the user perspective?

Yeah, it just rewrites the space-flag syntax to an equal-flag syntax for the token flag behind the scenes. User passes their arguments the same way.

**Comment by lesam** [comment]
The problem would be if people write something like:

```
./influx config create -n local_2 -u http://localhost:8086/ -t -a
Active  Name    URL                     Org
        local_2 http://localhost:8086/
```

The `-a` gets interpreted as a token when it was intended as an `active` marker, which is weird.

On the whole, I think this is better than the alterative though.

Maybe we could have a stderr print like `warning: -t -a interpretted as -t=-a, consider using -t=-a syntax when tokens may start with a hyphen`.

@samhld thoughts?

**Comment by samhld** [comment]
@lesam I think that works if this is the path we're taking. What was the last sentiment on just changing the token generation server-side to disallow this type of token. Do you think it's reasonable users would bring their own token starting with a hyphen? I suppose we'd have to support that even if it's an edge case, if we're going to support "bring your own token" at all.

File: cmd/influx/main.go [hunk 2]
```diff
@@ -91,9 +93,42 @@ func newApp() cli.App {
...
+	copy(newArgs, args)
+	for i, arg := range newArgs {
+		switch arg {
+		case "--token", "-t":
+			// if last element, this will be invalid later
+			if i == len(args)-1 {
+				break
+			}
+			if strings.HasPrefix(newArgs[i+1], "-") {
+				color.HiYellow("warning: %s %s interpreted as %s=%s, consider using %s=%s syntax when tokens start with a hyphen",
```
Comment (lesam) [inline cmd/influx/main.go:110]: Can you use explicit argument indexing in the format string here?

File: cmd/influx/main.go [hunk 3]
```diff
@@ -91,9 +93,42 @@ func newApp() cli.App {
...
+					newArgs[i], newArgs[i+1],
+					newArgs[i], newArgs[i+1],
+					newArgs[i], newArgs[i+1],
+				)
+			}
+			newArgs[i] = strings.Join(newArgs[i:i+2], "=")
+			// if there are 2+ elements after this
+			if len(newArgs) > i+2 {
+				newArgs = append(newArgs[:i+1], newArgs[i+2:]...)
```
Comment (lesam) [inline cmd/influx/main.go:119]: I don't think you need the 'if' here - `newArgs` would be an empty slice if `len(newArgs) == i+2`

File: cmd/influx/main.go [hunk 4]
```diff
@@ -91,9 +93,42 @@ func newApp() cli.App {
 	}
 }
 
+// This creates a new slice and replaces `-t "-FOO-TOKEN"` with `-t=-FOO-TOKEN`
+// This is necessary to do because the command line arg:
+//  `-t "-FOO-TOKEN"`` will be parsed as two separate flags instead of a flag and token value.
+func replaceTokenArg(args []string) []string {
```
Comment (lesam) [inline cmd/influx/main.go:99]: Can you write a unit test for `replaceTokenArg` ?

**Review by lesam — CHANGES_REQUESTED** [review]

File: cmd/influx/main.go (hunk 2 above)
Comment (candrewlee14) [reply cmd/influx/main.go:110]: TIL!

**Review by lesam — APPROVED** [review]

**Pull request merged**